### ISSUE TITLE
Use external functions to generate query string parameters doesn't work after tavern-1.12.2

### ISSUE THREAD

**Issue by DonghanYang** [taverntesting/tavern#632]
```yaml
name: Request trade analysis for a date range
request:
  url: "{tavern.env_vars.APP_URL}/v2/price"
  method: GET
  verify: false
  params:
    $ext:
      function: api.test.tavern.utils:generate_multi_date_request
  headers:
    content-type: application/json
response:
  status_code: 200
  verify_response_with:
    function: api.test.tavern.utils:validate_ta_data
```
We have a test case that looks like this and use an external Python function to generate the query string parameters. This used to work before `tavern-1.12.2`. We didn't change any code but only bumped the tavern version so the issue should be replicable using a dummy example as well. Not sure if this is related to the `pykwalify` version.

**Comment by kstilwell** [comment]
I took a look at this (since I was trying to call an external function from headers as well) and the issue is that any JSON data that is under params is being url_encoded by `get_request_args` (https://github.com/taverntesting/tavern/blob/f18aed6636541a75a10e84374f185c7a30713c04/tavern/_plugins/rest/request.py#L120) which breaks `update_from_ext` being able to process `$ext` tags for params.

**Comment by michaelboulton** [comment]
This should now be fixed in 1.13.1
